workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ScreenViewFactoryFinder.kt

```diff
@@ -57,18 +57,20 @@ public interface ScreenViewFactoryFinder {
     @Suppress("UNCHECKED_CAST")
     return (entry as? ScreenViewFactory<ScreenT>)
       ?: (rendering as? AndroidScreen<*>)?.viewFactory as? ScreenViewFactory<ScreenT>
-      ?: (rendering as? AsScreen<*>)?.let { AsScreenViewFactory as ScreenViewFactory<ScreenT> }
+      ?: (rendering as? AsScreen<*>)?.let {
+        AsScreenViewFactory(it, environment) as ScreenViewFactory<ScreenT>
```

Comment:
- [ ] `AsScreenViewFactory` should be updated to use param type correctly, see `NamedScreenViewFactory` and `EnvironmentScreenViewFactory`.

Reply:
Or maybe those two can be `object` instead of `fun`

Reply:
Oh, it's different b/c it wraps any old legacy screen.